### PR TITLE
fix(source-braintree): Fixes subscription data by @agpapa

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

Copy of @AGPapa's PR to address bugs in source-braintree. #40693

## How

@AGPapa pushed up a pin commit here, let's see if this helps with pyairbyte tests and acceptance tests. If these are green, cherry pick my commit and we'll merge your PR and close this. Or the other way around, you still get the credit. 

Original PR: 

## What
For several columns in the subscriptions stream the type is incorrectly "date-time" when it should be "date". This causes syncs involving subscriptions to fail with this message: `invalid type; expected datetime, string, bytes, int or float (type=type_error)`

This has been brought up in at least two issues:

https://github.com/airbytehq/airbyte/issues/15547
https://github.com/airbytehq/airbyte/issues/14186

## How
I switched the types to 'dates' instead of 'date-times' and it works! I also added in a missing column to the subscriptions stream - billing_period_end_date.

I tested this by running a connection from braintree's sandbox to a local CSV file. It failed before the change and works after it.

I also switched it from using the Dockerfile approach to using poetry instead (as directed by the test suite).

## User Impact
Users with subscription data will be able to sync it using airbyte. Users without subscription data should be unaffected, although there might be a datatype change in their empty subscription tables.

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by natikgadzhi** [comment]
Yo @AGPapa, I figured out the pyairbyte situation.

```
[tool.poetry.scripts]
source-exchange-rates = "source_braintree.run:run"
```

Wrong executable name! 

There's also a bug in pyairbyte installer that fails to interpolate variable ;)

**Comment by natikgadzhi** [comment]
@AGPapa if this works, mind if I just merge this instead of cherrypicking back?

**Comment by AGPapa** [comment]
> @AGPapa if this works, mind if I just merge this instead of cherrypicking back?


Go for it!

File: airbyte-integrations/connectors/source-braintree/pyproject.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ airbyte-cdk = "^0.50"
 braintree = "^4.28.0"
 
 [tool.poetry.scripts]
-source-exchange-rates = "source_braintree.run:run"
```
Comment (AGPapa) [inline airbyte-integrations/connectors/source-braintree/pyproject.toml]: 🤦 

oof - I copied this file from that source, missed the rename here

Comment (natikgadzhi) [reply airbyte-integrations/connectors/source-braintree/pyproject.toml]: Bruh ☠️

**Comment by natikgadzhi** [comment]
Automerge is on. @ChristoGrab, it's all yours!

**Review by ChristoGrab — APPROVED** [review]

**Pull request merged**